plugins/c9.ide.ace.keymaps/cli.js

```diff
@@ -357,26 +473,37 @@ define(function(require, exports, module) {
                 cmd += last.value;
             cmd = cmd.substr(1).trim();
             var args = cmd.split(/\s+/);
-            cmd = args[0];
+            var firstCmd = args[0];
         
-            if (this.commands[cmd]) {
-                cmd = this.commands[cmd];
+            if (this.commands[firstCmd]) {
+                cmd = this.commands[firstCmd];
                 if (typeof cmd == "string")
                     return commands.exec(cmd, null, { argv: args });
                 else if (typeof cmd == "function")
                     return cmd(ed, { argv: args });
                 else if (cmd.exec)
                     return cmd.exec(ed, { argv: args });
             }
-            else if (commands.commands[cmd]) {
-                commands.exec(cmd, null, { argv: args });
+            else if (commands.commands[firstCmd]) {
+                commands.exec(firstCmd, null, { argv: args });
             }
             else if (/^\d+$/.test(cmd)) {
                 ed.gotoLine(cmd, 0, true);
             }
             else {
-                ed.cmdLine.setTimedMessage("Vim command '" 
-                    + cmd + "' not implemented.", 1500);
+                for (var key in this.reCommands) {
```

Comment:
are there any other reCommands that we'll need to add here in the future

Reply:
All commands with range (Delete, Yank, Read/Save) and i hope add shell commands